pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,56 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"testing"
+	"time"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zone   []configv1.DNSZone
+		result reconcile.Result
+		expect (operatorv1.ConditionStatus)
+	}{
+		{
+			name: "check status",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zone: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			result: reconcile.Result{RequeueAfter: 10 * time.Second},
+			expect: operatorv1.ConditionFalse,
+		},
```

Comment:
You could add a test case with empty `zones`.  If you do, I suggest changing the type of `expect` to a slice of conditions: `[]operatorv1.ConditionStatus`, and asserting that `r.publishRecordToZones` returns the expected number of conditions with the expected statuses.

Reply:
done